tensorrt_llm/_torch/custom_ops/cute_dsl_custom_ops.py

```
@@ -5707,17 +5708,96 @@ def _(
-        ``min_blocks_per_mp`` (ptxas ``__launch_bounds`` cap) and
-        ``enable_warp_parallel_reduce`` from input shape + dtype.
-
-        Production knobs (tuning parameters) are fixed: ``enable_unroll_4``
-        and ``enable_phase3_unroll`` always on, ``use_constant_hint`` off.
+        Owns the JIT cache and the (T, V, min_blocks_per_mp,
+        warp_parallel_reduce) heuristic. ``forward()`` dispatches three
+        paths from ``(counters, order_row)`` — see its docstring. All
+        share :meth:`_pick_tuning`; only the compiled kernel class and
+        launch signature differ.
         """
         kernel_cache: dict = {}
```

Ruff (0.15.20)

[warning] Mutable default value for class attribute

(RUF012)